### ISSUE TITLE
Marvel Land (USA) - black screen at game over

### ISSUE THREAD

**Issue by mali2297** [MiSTer-devel/Genesis_MiSTer#157]
When losing all lives in area 2, one expects a game over screen with a password and an option to continue. Instead, one gets a black screen. The issue does not occur in area 1 where one gets a plain game over screen without password as expected.

Marvel Land (USA).md
CRC32: cd7eeeb7

**Comment by mali2297** [comment]
Some further findings:
If one opens the core menu and changes the region from US to PAL then the expected game over screen suddenly appears. The screen remains if one changes the region back to US again and one can choose to continue the game.
The issue is unpredictable as it does not always occur. In one trial, the game over screen appears as expected. In the next trial, one might get a black screen. However, the fix of switching regions always work.

**Comment by bootsector** [comment]
Looks like the Streets of Rage "The Simpsons" hack where we need to switch the region at some point in order to get over a black screen as well.

**Comment by birdybro** [comment]
Yeah just recreated this on latest release. However, it's not just switching the region that fixes this. Also, just resetting in the OSD (which just is like pressing the reset button on the system) changes this. Possibly related to https://github.com/MiSTer-devel/Genesis_MiSTer/issues/151 as a simple reset fixes that problem too. Is it possible that the initial state of the system's RAM is not the same as original hardware and this could be causing a few issues?

**Comment by bootsector** [comment]
Anyone has compiled https://github.com/MiSTer-devel/Genesis_MiSTer/commit/8f3ab0fb7425ac4d0f711c66e18b1577cb9d103f so we can see if it fixes this as well?

**Comment by birdybro** [comment]
I'll test it today.

**Comment by birdybro** [comment]
Using https://github.com/MiSTer-devel/Genesis_MiSTer/commit/040d88e0864e86d3d5afc7c6f384e9b8614caed3 actually, since that is out now...

Load game + SDRAM -> level 2 gameover = Works (this didn't before, it went black screen)
Load game + DDR3 -> level 2 gameover = Works (same as before)

Seems like it's good now! Just release update needs to happen at some point to distribute it out.

**Comment by mali2297** [comment]
I am sorry to report that the issue remains both in the preview build and today's official release (20210105). 

I again got stuck at a black screen after my first game over in area 2. As I wrote in a previous post, the issue is not entirely predictable so I suggest doing more than one test.

**Comment by birdybro** [comment]
Yeah, I might not have tested thoroughly now that i think about it, since I didn't restart the mister from cold each time, maybe something is lingering and not giving the same result. This time I did the following.

Cold reboot mister (or rather power it on the firs ttime). Go into genesis core, set genesis core to the desired RAM setting i'm testing. Save that core settings. Do a cold reboot, to make sure any lingering memory stuff or whatever else is cleaned up. Then I load genesis core, load the game.

With Auto = Fails first time. Toggling region brings the continue menu up as before. Same result after updates as reported.
Cold reboot, do it again, this time it didn't fail. Weird.  
Did a regular reset in the core (like genesis reset), started the game, accidentally paused right away, and it locked up and won't unpause.  
Reset the game again, pause freezes up the game at the start again, so not going to do that anymore. Something is wacky here.  
Reset the game again, got to level 2 didn't need to pause, got game over, black screen again.

Set to SDRAM = same behavior as above, so Auto is setting it to SDRAM and SDRAM differences to DDR3 are part of the problem. DDR3 does work just fine after 3 runs.

I know one thing, after repeating this about 8 times in a row, I don't want to hear that music again :P 

Conclusion: Not resolved, shouldn't be closed, and it isn't just toggling region that changes behavior, toggling ROM Storage in miscellaneous changes behavior as well. DDR3 prevents the bug, SDRAM shows the bug occur.

**Comment by bootsector** [comment]
It seems to get The Simpsons SoR2 hack fixed though. However I'm getting lack of sync in screen transitions where it used to lock in a black screen before the fix. Too bad it doesn't fix this game.

**Comment by HiroiSekai** [comment]
I want to add that I had this issue as well. Ran the game for a speedrun race and got a black screen at the game over. Ended the race and had to start over.

Another weird issue is that if you try and pause at certain intervals (I notice it most right when you gain control for the first time, but it happens in other places too), the game essentially hardlocks and you can't do anything. Only thing that still responds is the MiSTer OSD so you're forced to reset.

**Comment by HiroiSekai** [comment]
Just wanted to note this is still happening on the newest version of the core. I speedrun this game on occasion and I even had the game black screen on me during a speedrun marathon race event, killing one of the best-of-3 races.

Would love to hear if there was any reports of this getting replicated and/or looked at. Thank you.

**Comment by birdybro** [comment]
Yup. However it is resolved using the Sega 32x core, so it is like another issue resolved by the rewritten vdp, another bus arbitration bug.

**Comment by sorgelig** [comment]
I think most problem are coming from bus arbiter. VDP more or less correct i think.

**Comment by HiroiSekai** [comment]
Thanks, glad to know it's a known resolution. Will this be implemented into the Genesis core? I don't think you can run Genesis games on the 32X core directly, unless I'm missing something.

**Comment by birdybro** [comment]
You can, but many games have bugs because it's not really a Genesis-compatible core fully. You just have to rename the file extension.

**Comment by HiroiSekai** [comment]
Thank you, I'll give it a whirl and keep the bug reporting note in mind. Just to recheck, is this something that can be implemented into the Genesis core or should I keep testing this on the S32X core for the immediate future?

**Comment by birdybro** [comment]
@sorgelig This also gets resolved by the core you ported from @nukeykt
